index_configuration_tool/main.py

```diff
@@ -77,10 +78,13 @@ def validate_plugin_config(config: dict, key: str):
     plugin_config = supported_endpoint[1]
     if HOSTS_KEY not in plugin_config:
         raise ValueError("No hosts defined for endpoint: " + supported_endpoint[0])
-    if USER_KEY in plugin_config and PWD_KEY not in plugin_config:
-        raise ValueError("Invalid auth configuration (no password for user) for endpoint: " + supported_endpoint[0])
-    elif PWD_KEY in plugin_config and USER_KEY not in plugin_config:
-        raise ValueError("Invalid auth configuration (Password without user) for endpoint: " + supported_endpoint[0])
+    if not plugin_config.get(DISABLE_AUTH_KEY, False):
+        if USER_KEY in plugin_config and PWD_KEY not in plugin_config:
+            raise ValueError("Invalid auth configuration (no password for user) for endpoint: " +
+                             supported_endpoint[0])
+        elif PWD_KEY in plugin_config and USER_KEY not in plugin_config:
+            raise ValueError("Invalid auth configuration (Password without user) for endpoint: " +
+                             supported_endpoint[0])
```

Comment:
I want to make sure I understand the logic in this section:
1. Disable Auth key is present: any other combo is okay
2. Disable Auth key is _not_ present AND both user and pass are present: we're good, we have auth 
3. Disable Auth key is _not_ present AND only one of user and pass are present: not good, handled by this logic
4. Disable Auth key is _not_ present AND neither user nor pass is present: ??? Is this okay? Or is it handled elsewhere?

Reply:
Good catch on 4 - if the disable auth key is not present, the logic in 2 and 3 should apply. I'll fix this in a follow-up PR

Reply:
Ah, sorry I didn't catch that, Kartik. Thanks Mikayla!

Reply:
https://github.com/opensearch-project/opensearch-migrations/pull/222